test/e2e/specs/specs-playwright/wp-signup__screenshot.ts

```diff
@@ -0,0 +1,299 @@
+/**
+ * @group calypso-release
+ */
+import fs from 'fs';
+import {
+	ChangeUILanguageFlow,
+	DataHelper,
+	DomainSearchComponent,
+	LoginPage,
+	CartCheckoutPage,
+	setupHooks,
+	UserSignupPage,
+	SignupPickPlanPage,
+	CloseAccountFlow,
+	BrowserManager,
+} from '@automattic/calypso-e2e';
+import archiver from 'archiver';
+import FormData from 'form-data';
+import fetch from 'node-fetch';
+import { Page } from 'playwright';
+import type { LanguageSlug } from '@automattic/languages';
+
+const selectors = {
+	isWhiteLogin: '.is-section-login.is-white-login',
+	isBlueLogin: '.is-section-login:not( .is-white-login )',
+	isWhiteSignup: '.is-white-signup .is-section-signup',
+	isBlueSignup: '.is-section-signup:not( .is-white-signup )',
+};
+
+describe( DataHelper.createSuiteTitle( 'Signup: WordPress.com Free' ), function () {
+	const inboxId = DataHelper.config.get( 'inviteInboxId' ) as string;
+	const username = `e2eflowtestingfree${ DataHelper.getTimestamp() }`;
+	const email = DataHelper.getTestEmailAddress( {
+		inboxId: inboxId,
+		prefix: username,
+	} );
+	const signupPassword = DataHelper.config.get( 'passwordForNewTestSignUps' ) as string;
+	const blogName = DataHelper.getBlogName();
+
+	let page: Page;
+	let selectedDomain: string;
+
+	setupHooks( ( args ) => {
+		page = args.page;
+	} );
+
+	describe( 'Signup', function () {
+		jest.setTimeout( 1000000 );
+		let cartCheckoutPage: CartCheckoutPage;
+		const magnificientNonEnLocales = [
+			'pt-br',
+			'fr',
+			'es',
+			'de',
+			'he',
+			'ja',
+			'it',
+			'nl',
+			'ru',
+			'tr',
+			'id',
+			'zh-cn',
+			'zh-tw',
+			'ko',
+			'ar',
+			'sv',
+		];
+
+		it( 'Screenshot blue login page in desktop viewport, en and Mag-16 locales', async function () {
+			const loginPage = new LoginPage( page );
+			for ( const locale of [ 'en', ...magnificientNonEnLocales ] ) {
```

Comment:
As long as this script is not required to run as part of regular calypso deployment process, I have no issues with the `for...of` loop within a test step.

However, if this script is expected to run as part of every calypso deployment, then this loop leads to issues with TeamCity being unable to break out the individual steps taken (eg. Screenshot blue login page (en)).

In that case, I'd like to request the use of `it.each` parametrization for this step and all other steps that iterate through the locale.

Reply:
I expect this script to run every time a change is made to signup, login, or checkout file paths. So, it won't run on  every Calypso deployment. So is it fine to retain the `for ... of` loop, or do you still suggest to use `it..each`?

Reply:
Generally, using Jest's built-in parallelization tool is preferred over using `for ... of` loop, so I would prefer the following:

```
it.each( ['en', ...magnificientNonEnLocales ])( 'Screenshot blue login page in desktop viewport for %s' )
```

which would generate something like

```
'Screenshot blue login page in desktop viewport for en' 
'Screenshot blue login page in desktop viewport for fr'
'Screenshot blue login page in desktop viewport for de'
```

and so on.